features: fix coding style

0xc0170 opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->

AStyle run on all features that we have (ignored folders are not being checked). This needs further work (we might not have ignore folders up to date, or some features might need update upstream). Let's review what can be updated and how. 

 I'll add reviewers with questions about the features and its style (some might need some edits).


### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

0xc0170 committed:
cellular: fix coding style

0xc0170 committed:
device key: fix coding style

0xc0170 committed:
lorawan: fix coding style

0xc0170 committed:
storage: fix astyle coding style

0xc0170 committed:
lwipstack: fix astyle coding style

0xc0170 committed:
astyleignore: update nanostack paths

0xc0170 committed:
feature deprecated: fix astyle coding style

0xc0170 committed:
lorawan: fix astyle coding style

0xc0170 committed:
netsocket: fix astyle coding style

0xc0170 commented:
> travis-ci/astyle — Passed, 66 files (-480 files)

Need to find these 66 files 👀

0xc0170 commented:
Initial feedback for files changed here:

@ARMmbed/mbed-os-storage 
@ARMmbed/mbed-os-ipcore 
@ARMmbed/mbed-os-pan 
@ARMmbed/mbed-os-wan 

Please review your features here and let me know what can be run through astyle (I used .astyleignore paths to ignore), so if files here are changed, means I need to rerun it with updated ignored folders.

pan- commented on features/FEATURE_BLE/ble/ArrayView.h:
```diff
@@ -102,8 +103,8 @@ struct ArrayView {
      */
-    ArrayView(T (&elements)[Size]):
-		_array(elements) { }
+    ArrayView(T(&elements)[Size]):
```
Astyle nonsense: 

```suggestion
    ArrayView(T (&elements)[Size]):
```

0xc0170 replied:
This is known.

We need to turn off astyle for any template magic, similar in Span class and others. I'll fix that . Thanks for the pointers !

Main question - is there any BLE code that should be excluded apart from this errors?

pan- commented on features/FEATURE_BLE/ble/ArrayView.h:
```diff
@@ -203,7 +204,7 @@ struct ArrayView<T, ARRAY_VIEW_DYNAMIC_SIZE> {
      */
     template<size_t Size>
-    ArrayView(T (&elements)[Size]):
+    ArrayView(T(&elements)[Size]):
```
Astyle nonsense: 
```suggestion
    ArrayView(T (&elements)[Size]):
```

pan- commented on features/FEATURE_BLE/ble/ArrayView.h:
```diff
@@ -333,7 +334,7 @@ bool operator!=(const ArrayView<T, LhsSize>& lhs, const ArrayView<T, LhsSize>& r
  */
 template<typename T, size_t Size>
-ArrayView<T, Size> make_ArrayView(T (&elements)[Size])
+ArrayView<T, Size> make_ArrayView(T(&elements)[Size])
```
Astyle nonsense: 
```suggestion
ArrayView<T, Size> make_ArrayView(T (&elements)[Size])
```

pan- commented on features/FEATURE_BLE/ble/ArrayView.h:
```diff
@@ -389,7 +390,7 @@ ArrayView<T> make_ArrayView(T* array_ptr, size_t array_size)
  */
 template<typename T, size_t Size>
-ArrayView<const T, Size> make_const_ArrayView(T (&elements)[Size])
+ArrayView<const T, Size> make_const_ArrayView(T(&elements)[Size])
```
Astyle nonsense: 
```suggestion
ArrayView<const T, Size> make_const_ArrayView(T (&elements)[Size])
```

pan- commented on features/FEATURE_BLE/ble/BLE.h:
```diff
@@ -471,7 +473,8 @@ class BLE
         BLEProtocol::AddressType_t *typeP, BLEProtocol::AddressBytes_t address
-    ) {
+    )
+    {
```
Why ?

0xc0170 replied:
`{` for functions is not attached

pan- commented on features/FEATURE_BLE/ble/BLETypes.h:
```diff
@@ -76,7 +76,8 @@ struct attribute_handle_range_t {
         const attribute_handle_range_t &lhs, const attribute_handle_range_t &rhs
-    ) {
+    )
+    {
```
I'm not sure about that one, the opening brace is already on a new line

0xc0170 replied:
Why this is not formatted as (80 character limit I bet), this should be it:

```
   friend bool operator==(const attribute_handle_range_t &lhs, const attribute_handle_range_t &rhs)
```

I 'll review our default indentation options

pan- commented on features/FEATURE_BLE/ble/BLETypes.h:
```diff
@@ -90,7 +91,8 @@ struct attribute_handle_range_t {
         const attribute_handle_range_t &lhs, const attribute_handle_range_t &rhs
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/BLETypes.h:
```diff
@@ -112,7 +114,8 @@ struct attribute_handle_range_t {
     attribute_handle_t end
-) {
+)
+{
```
same as above.

pan- commented on features/FEATURE_BLE/ble/BLETypes.h:
```diff
@@ -662,14 +685,16 @@ class phy_set_t {
         bool phy_2m,
         bool phy_coded
-    ) {
+    )
```
Same as above.

pan- commented on features/FEATURE_BLE/ble/CharacteristicDescriptorDiscovery.h:
```diff
@@ -111,8 +111,8 @@ class CharacteristicDescriptorDiscovery {
-    typedef FunctionPointerWithContext<const DiscoveryCallbackParams_t*>
-        DiscoveryCallback_t;
+    typedef FunctionPointerWithContext<const DiscoveryCallbackParams_t *>
+    DiscoveryCallback_t;
```
Please keep the indentation as the expression is not complete.

0xc0170 replied:
https://sourceforge.net/p/astyle/tickets/37/ - this is the bug there, will check if there is any fix

pan- commented on features/FEATURE_BLE/ble/CharacteristicDescriptorDiscovery.h:
```diff
@@ -128,8 +128,8 @@ class CharacteristicDescriptorDiscovery {
-    typedef FunctionPointerWithContext<const TerminationCallbackParams_t*>
-        TerminationCallback_t;
+    typedef FunctionPointerWithContext<const TerminationCallbackParams_t *>
+    TerminationCallback_t;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/DiscoveredCharacteristic.h:
```diff
@@ -551,8 +552,9 @@ class DiscoveredCharacteristic {
-        const DiscoveredCharacteristic& lhs, const DiscoveredCharacteristic& rhs
-    ) {
+        const DiscoveredCharacteristic &lhs, const DiscoveredCharacteristic &rhs
+    )
```
Again, I'm not so sure to understand why the opening brace should be on a new line.

pan- commented on features/FEATURE_BLE/ble/DiscoveredCharacteristic.h:
```diff
@@ -571,8 +573,9 @@ class DiscoveredCharacteristic {
-    ) {
+        const DiscoveredCharacteristic &lhs, const DiscoveredCharacteristic &rhs
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/DiscoveredService.h:
```diff
@@ -109,7 +110,8 @@ class DiscoveredService {
         GattAttribute::Handle_t endHandleIn
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/DiscoveredService.h:
```diff
@@ -129,7 +131,8 @@ class DiscoveredService {
         GattAttribute::Handle_t endHandleIn
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/DiscoveredService.h:
```diff
@@ -147,8 +150,9 @@ class DiscoveredService {
-    ) {
-         uuid.setupLong(longUUID, order);
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -731,7 +731,7 @@ class Gap {
      */
     typedef FunctionPointerWithContext<const AdvertisementCallbackParams_t *>
-        AdvertisementReportCallback_t;
+    AdvertisementReportCallback_t;
```
Please keep the indentation as the expression is not complete.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1031,10 +1031,10 @@ class Gap {
 
     static const PeripheralPrivacyConfiguration_t
-        default_peripheral_privacy_configuration;
+    default_peripheral_privacy_configuration;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1031,10 +1031,10 @@ class Gap {
 
     static const CentralPrivacyConfiguration_t
-        default_central_privacy_configuration;
+    default_central_privacy_configuration;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1064,39 +1064,39 @@ class Gap {
      */
     typedef CallChainOfFunctionPointersWithContext<TimeoutSource_t>
-        TimeoutEventCallbackChain_t;
+    TimeoutEventCallbackChain_t;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1064,39 +1064,39 @@ class Gap {
      */
     typedef FunctionPointerWithContext<const ConnectionCallbackParams_t *>
-        ConnectionEventCallback_t;
+    ConnectionEventCallback_t;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1064,39 +1064,39 @@ class Gap {
      */
     typedef CallChainOfFunctionPointersWithContext<const ConnectionCallbackParams_t *>
-        ConnectionEventCallbackChain_t;
+    ConnectionEventCallbackChain_t;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1064,39 +1064,39 @@ class Gap {
-    typedef FunctionPointerWithContext<const DisconnectionCallbackParams_t*>
-        DisconnectionEventCallback_t;
+    typedef FunctionPointerWithContext<const DisconnectionCallbackParams_t *>
+    DisconnectionEventCallback_t;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1064,39 +1064,39 @@ class Gap {
-    typedef CallChainOfFunctionPointersWithContext<const DisconnectionCallbackParams_t*>
-        DisconnectionEventCallbackChain_t;
+    typedef CallChainOfFunctionPointersWithContext<const DisconnectionCallbackParams_t *>
+    DisconnectionEventCallbackChain_t;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1118,7 +1118,7 @@ class Gap {
      */
     typedef CallChainOfFunctionPointersWithContext<const Gap *>
-        GapShutdownCallbackChain_t;
+    GapShutdownCallbackChain_t;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1144,7 +1144,8 @@ class Gap {
             Phy_t rxPhy
-        ) {
+        )
+        {
```
see other comments.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1177,7 +1178,8 @@ class Gap {
             Phy_t rxPhy
-        ) {
+        )
+        {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1237,7 +1239,8 @@ class Gap {
         const BLEProtocol::AddressBytes_t address
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1261,7 +1264,8 @@ class Gap {
         BLEProtocol::AddressBytes_t address
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1377,7 +1381,8 @@ class Gap {
         const GapScanningParams *scanParams
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1418,7 +1423,8 @@ class Gap {
         const GapScanningParams *scanParams
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1481,9 +1488,10 @@ class Gap {
+        const PhySet_t *txPhys,
+        const PhySet_t *rxPhys
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1511,10 +1519,11 @@ class Gap {
         CodedSymbolPerBit_t codedSymbol
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1533,7 +1542,8 @@ class Gap {
         Handle_t connectionHandle, DisconnectionReason_t reason
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1599,7 +1610,8 @@ class Gap {
         const ConnectionParams_t *params
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1623,7 +1635,8 @@ class Gap {
         const ConnectionParams_t *params
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -1745,7 +1759,8 @@ class Gap {
         const int8_t **valueArrayPP, size_t *countP
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2117,7 +2132,8 @@ class Gap {
         GapAdvertisingData::DataType type, const uint8_t *data, uint8_t len
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2158,7 +2174,8 @@ class Gap {
         GapAdvertisingData::DataType type, const uint8_t *data, uint8_t len
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2214,7 +2231,8 @@ class Gap {
         GapAdvertisingData::DataType type, const uint8_t *data, uint8_t len
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2277,11 +2296,12 @@ class Gap {
         bool activeScanning = false
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2277,11 +2296,12 @@ class Gap {
         if (((rc = _scanningParams.setInterval(interval)) == BLE_ERROR_NONE) &&
-            ((rc = _scanningParams.setWindow(window))     == BLE_ERROR_NONE) &&
-            ((rc = _scanningParams.setTimeout(timeout))   == BLE_ERROR_NONE)) {
+                ((rc = _scanningParams.setWindow(window))     == BLE_ERROR_NONE) &&
```
No need for extra indentations.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2299,13 +2319,14 @@ class Gap {
-            scanningParams.getTimeout(),
-            scanningParams.getActiveScanning()
-        );
+                   scanningParams.getInterval(),
```
Extra indentations not needed.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2414,7 +2435,8 @@ class Gap {
         void (*callback)(const AdvertisementCallbackParams_t *params)
-    ) {
+    )
+    {
```
Why would we lost that space ?

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2447,7 +2469,8 @@ class Gap {
         void (T::*callbackMember)(const AdvertisementCallbackParams_t *params)
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2527,7 +2551,8 @@ class Gap {
         const PeripheralPrivacyConfiguration_t *configuration
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2541,7 +2566,8 @@ class Gap {
         PeripheralPrivacyConfiguration_t *configuration
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2554,7 +2580,8 @@ class Gap {
         const CentralPrivacyConfiguration_t *configuration
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/Gap.h:
```diff
@@ -2568,7 +2595,8 @@ class Gap {
         CentralPrivacyConfiguration_t *configuration
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GapAdvertisingData.h:
```diff
@@ -768,11 +768,11 @@ class GapAdvertisingData
-            static_cast<const GapAdvertisingData*>(this)->findField(type)
-        );
+        return const_cast<uint8_t *>(
+                   static_cast<const GapAdvertisingData *>(this)->findField(type)
```
No.

pan- commented on features/FEATURE_BLE/ble/GapAdvertisingData.h:
```diff
@@ -798,11 +798,12 @@ class GapAdvertisingData
-    ) {
+        uint8_t *field
+    )
+    {
```
revert.

pan- commented on features/FEATURE_BLE/ble/GapAdvertisingData.h:
```diff
@@ -866,8 +867,9 @@ class GapAdvertisingData
-    ) {
+        uint8_t *field
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattAttribute.h:
```diff
@@ -325,27 +326,29 @@ class GattAttribute {
      * Security applied to the read operation.
      */
-    uint8_t _read_security: Security_t::size;
+uint8_t _read_security:
```
Aha, no!

pan- commented on features/FEATURE_BLE/ble/GattAttribute.h:
```diff
@@ -325,27 +326,29 @@ class GattAttribute {
      * Security applied to the write operation.
      */
-    uint8_t _write_security: Security_t::size;
+uint8_t _write_security:
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1558,7 +1560,8 @@ class GattCharacteristic {
         void (*callback)(GattWriteAuthCallbackParams *)
-    ) {
+    )
+    {
```
Please revert.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1578,7 +1581,8 @@ class GattCharacteristic {
         void (T::*member)(GattWriteAuthCallbackParams *)
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1594,7 +1598,8 @@ class GattCharacteristic {
         void (*callback)(GattReadAuthCallbackParams *)
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1615,7 +1620,8 @@ class GattCharacteristic {
         void (T::*member)(GattReadAuthCallbackParams *)
-    ) {
+    )
+    {
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1744,30 +1750,30 @@ class GattCharacteristic {
-            ),
-            getUpdateSecurityRequirement()
-        );
+                                            std::max(
```
No.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1744,30 +1750,30 @@ class GattCharacteristic {
                 return (needs_signing) ?
-                    SecurityManager::SECURITY_MODE_SIGNED_NO_MITM :
-                    SecurityManager::SECURITY_MODE_ENCRYPTION_NO_MITM;
+                       SecurityManager::SECURITY_MODE_SIGNED_NO_MITM :
```
No.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1744,30 +1750,30 @@ class GattCharacteristic {
                 return (needs_signing) ?
-                    SecurityManager::SECURITY_MODE_SIGNED_WITH_MITM :
-                    SecurityManager::SECURITY_MODE_ENCRYPTION_WITH_MITM;
+                       SecurityManager::SECURITY_MODE_SIGNED_WITH_MITM :
```
No.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1844,8 +1850,9 @@ class GattCharacteristic {
-    ) {
-        switch(mode) {
+    )
+    {
```
Please revert.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1891,26 +1898,27 @@ class GattCharacteristic {
      */
     FunctionPointerWithContext<GattReadAuthCallbackParams *>
-        readAuthorizationCallback;
+    readAuthorizationCallback;
```
Please revert as the expression is not complete.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1891,26 +1898,27 @@ class GattCharacteristic {
      */
     FunctionPointerWithContext<GattWriteAuthCallbackParams *>
-        writeAuthorizationCallback;
+    writeAuthorizationCallback;
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1891,26 +1898,27 @@ class GattCharacteristic {
      * receive updates
      */
-    uint8_t _update_security: SecurityRequirement_t::size;
+uint8_t _update_security:
```
No.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1944,15 +1952,16 @@ class ReadOnlyGattCharacteristic : public GattCharacteristic {
-        numDescriptors,
-        false
-    ) {
+            uuid,
```
No reason for the extra indentation.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -1987,14 +1996,15 @@ class WriteOnlyGattCharacteristic : public GattCharacteristic {
-        descriptors,
-        numDescriptors
-    ) {
+            uuid,
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -2029,14 +2039,15 @@ class ReadWriteGattCharacteristic : public GattCharacteristic {
-        descriptors,
-        numDescriptors
-    ) {
+            uuid,
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -2072,14 +2083,15 @@ class WriteOnlyArrayGattCharacteristic : public GattCharacteristic {
-        descriptors,
-        numDescriptors
-    ) {
+            uuid,
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -2116,15 +2128,16 @@ class ReadOnlyArrayGattCharacteristic : public GattCharacteristic {
-        numDescriptors,
-        false
-    ) {
+            uuid,
```
same as above.

pan- commented on features/FEATURE_BLE/ble/GattCharacteristic.h:
```diff
@@ -2161,14 +2174,15 @@ class ReadWriteArrayGattCharacteristic : public GattCharacteristic {
-        descriptors,
-        numDescriptors
-    ) {
+            uuid,
```
same as above.

pan- reviewed (changes requested):
Already too many comments and I'm not half way in for BLE.  Astyle is very dumb for some constructs.

pan- reviewed (dismissed):
Already too many comments and I'm not half way in for BLE.  Astyle is very dumb for some constructs.

AnttiKauppila reviewed (approved):
LoRaWAN and Cellular are fine

0xc0170 commented:
@AnttiKauppila Thank you!

@pan- I left one comment above, and will review all instances of BLE issues (If it makes easier, I can send separate PR). Let me know what would be the best approach for BLE code.

paul-szczepanek-arm commented:
Instead of making exceptions and fixing errors one by one maybe it would be worth it to come up with a set of astyle rules that you agree on? Most of the changes here could be fixed with a config change. Not all though but AStyle is open source and reasonably easy to modify. Vincent's function call and declaration style will probably require code modification to be applied and/or retained.

0xc0170 commented:
> Instead of making exceptions and fixing errors one by one maybe it would be worth it to come up with a set of astyle rules that you agree on? 

Already done via the config file we have - if you spot we are missing something, we can fix it. 
These issues that @pan- found are astyle shortcomings. See their docs, this is from their docs

>Artistic Style cannot always determine the usage of symbols with more than one meaning. For example an asterisk (*) can be multiplication, a pointer, or a pointer dereference. The "&" and "&&" symbols are a similar problem.
>
>If a symbol is being padded incorrectly, padding it manually may fix the problem. If it is still being padded incorrectly, then disabling the formatting may be necessary. To avoid having to use the "disable block" tags above, a single line disable is available.

> Most of the changes here could be fixed with a config change. Not all though but AStyle is open source and reasonably easy to modify.

I am not certain we can fix easily the above (*,&,&&).

From the experience, some code use different formatting than what we have in the config and astyle formats it wrongly anyway - manual update resolves it. I'll review all the review comments soon

paul-szczepanek-arm commented:
Some of these seem to be caused by unpad-paren not being clever enough. Sadly there is no unpad-paren-in

paul-szczepanek-arm commented:
The strength of astyle is that's easy to modify because it's so simple but the drawback of astyle is that it's so simple and doesn't understand the code it's styling.

There are some alternatives that are more langauge aware. When I got frustrated with astyle I switched to https://github.com/uncrustify/uncrustify. Using http://universalindent.sourceforge.net/ you could recreate the existing astyle config very quickly and then improve on it as uncrustify has more fine grained options and doesn't have the same bugs astyle has.

Apologies if this was already considered in the past.

0xc0170 commented:
Instead of focusing on the tool itself (thanks for the tip - haven't used it previously, we shall review uncrustify and clang format for future use. We need to work with the tools we have now), lets focus on the code and tools we have been using. 

Most of the review comments are related to the hard character limit - in this case 80 (lines are split). We haven't enforced this limit (our coding style suggested 120 but not enforced) thus would suggest reformat the code we identified - should be quick to do (run astyle and review as done here) and would align with the rest of the codebase (look at lorawan, had similar style issues). This should fix most of the issues that we identified here (multiline typedef, parameters to the object ctor and similar). If you got any other suggestions, lets discuss it in the new PR. I suggest removing BLE from here, create a new PR and reformat the code - done by @ARMmbed/mbed-os-pan team as they have better understanding of the codebase. 

I'll stash my changes for BLE just in case but will push an update to exclude BLE here in this pull request. Lets focus on the rest of features here

0xc0170 committed:
AStyle ignore: add BLE

BLE will be fixed via separate pull request as changes required are bigger.
We ignore BLE coding style for now.

0xc0170 commented:
Done , BLE feature excluded (added to the astyleignore file), will be fixed via separate PR.

0xc0170 committed:
rtos: fix coding style in the boot files

0xc0170 committed:
astyle ignore: add pithy code

0xc0170 committed:
tests: fix coding style

0xc0170 committed:
astyle ignore: add wifi/rf components

Will be reverted once upstream changes come to Mbed OS

0xc0170 committed:
tests: fix coding style

0xc0170 committed:
events: fix events coding style

0xc0170 committed:
astyle ignore: add lwip-sys

0xc0170 committed:
littlefs: fix coding style

0xc0170 commented:
Added also cleanup for the rest of the codebase (to check that ignore folders are also up to date, added new ignore folders) and this is now ready for review!

> travis-ci/astyle — Passed, 29 files (-516 files)

Most probably master and its ignore not up to date. I run astyle on entire codebase, all files should be resolved with this pull request (besides  what we merge in the meantime)

0xc0170 commented:
@ARMmbed/mbed-os-storage Please review features/storage changes
@ARMmbed/mbed-os-ipcore Please review nanostack/lwip/netsocket

The rest of changes are minimal update should be sufficient to be reviewed by @ARMmbed/mbed-os-maintainers

geky reviewed (approved):
I can't look too closely, but storage looks reasonable to me (not much advanced c++ there)

davidsaada reviewed (approved):
Storage changes look good. Thanks for this change.

0xc0170 commented:
Update:

- Rebased (one conflic resolved)
- netsockets/emac_drivers are ignored (added to the astyleignore and styling removed). They contain lot of 3rd party SDK files

0xc0170 commented:
/morph build

0xc0170 commented:
> travis-ci/astyle — Passed, 30 files (-515 files)

Once this one lands on master, I'll take care of the remaining files on master via new PR

cmonr commented:
Nodes dropped. Restarting.

/morph export-build

cmonr commented:
**Note**: This PR is now a part of a rollup PR (#8675).

Jenkins CI export nodes experienced many drops throughout the day causing false failures. In an attempt to get those PRs through CI, while keeping CI load low, several PRs have been bundled into a single rollup PR.

_If any more commits are made in this PR, this PR will remain open and have to go through CI on its own._

0xc0170 commented:
/morph build

0xc0170 commented:
I rebased to resolve one bad rebase, this is going alone (no rollup)

0xc0170 commented:
CI issues in the last failures, we are investigating

cc @OPpuolitaival

0xc0170 commented:
/morph build

cmonr commented:
Test timeout appears spurrious. Will retest when able.

cmonr commented:
/morph test

0xc0170 commented:
All green 🍏 💚  ready for integration!
